core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -479,6 +480,52 @@ private void applyUpdates(TableOperations underlyingOps) {
           // Cannot pass even with retry due to conflicting metadata changes. So, break the
           // retry-loop.
           throw new PendingUpdateFailedException(e);
+
+        } catch (ValidationException e) {
+          if (PropertyUtil.propertyAsBoolean(
+              base.properties(),
+              TableProperties.ROLLBACK_COMPACTION_ON_CONFLICTS_ENABLED,
+              TableProperties.ROLLBACK_COMPACTION_ON_CONFLICTS_ENABLED_DEFAULT)) {
+
+            // use refreshed metadata
+            this.base = underlyingOps.current();
+            this.current = underlyingOps.current();
+            Long rollbackToSnapshotId = current.currentSnapshot().parentId();
+            long currentSnapshotId = current.currentSnapshot().snapshotId();
+            boolean updatesAppliedSuccessfully = false;
+
+            while (rollbackToSnapshotId != null
+                && !updatesAppliedSuccessfully
+                && PropertyUtil.propertyAsBoolean(
```

Comment:
I discussed with you a bit offline as well, I think instead of introducing a new property of `is-compacted`, it seems like we can rollback whenever we have conflict and `current.snapshot(currentSnapshotId).equals(DataOperations.REPLACE)`? Which applies to both compaction and rewrite manifests and feels more generic. Is there any risk of doing that way?

Reply:
>  I discussed with you a bit offline as well, I think instead of introducing a new property of is-compacted, it seems like we can rollback whenever we have conflict and current.snapshot(currentSnapshotId).equals(DataOperations.REPLACE)

Apologies, I missed that. I think it makes sense to use snapshot#operation to identify if it's a compaction. Updated the PR & description.